pkg/controller/controller.go

```diff
@@ -314,46 +325,63 @@ func (c *RolloutController) adjustStatefulSetsGroupReplicas(ctx context.Context,
 	// are up-to-date on the models. If we modify the StatefulSet, they no longer are.
 	for _, sts := range sets {
 		currentReplicas := *sts.Spec.Replicas
+		reason := "to match leader"
 		desiredReplicas, err := desiredStsReplicas(groupName, sts, sets, c.logger)
 		if err != nil {
 			return false, err
 		}
 
-		if desiredReplicas > currentReplicas {
```

Comment:
From a design perspective, I’m not much convinced of modifying adjustStatefulSetsGroupReplicas() because it adds complexity mixing 2 different ways of scaling.

I would like to propose you an alternative design that should be easy to do and IMO cleaner:
Rename adjustStatefulSetsGroupReplicas() to adjustStatefulSetsGroupReplicasToFollowLeader()
Add your new logic into a new function adjustStatefulSetsGroupReplicasToMirrorResource()
Then create adjustStatefulSetsGroupReplicas() which calls adjustStatefulSetsGroupReplicasToFollowLeader() first, and if it returns false it will call adjustStatefulSetsGroupReplicasToMirrorResource()

This way we keep the 2 ways of scaling separated.

WDYT?